src/adaptermanager.js

```diff
@@ -194,22 +202,26 @@ exports.callBids = (adUnits, bidRequests, addBidResponse, doneCb) => {
       }
     }
   }
-  let ajax = ajaxBuilder(bidRequests[0].timeout);
-  bidRequests.forEach(bidRequest => {
-    bidRequest.start = new Date().getTime();
-    // TODO : Do we check for bid in pool from here and skip calling adapter again ?
-    const adapter = _bidderRegistry[bidRequest.bidderCode];
-    if (adapter) {
-      utils.logMessage(`CALLING BIDDER ======= ${bidRequest.bidderCode}`);
-      events.emit(CONSTANTS.EVENTS.BID_REQUESTED, bidRequest);
-      bidRequest.doneCbCallCount = 0;
-      let done = doneCb(bidRequest.bidderRequestId);
-      adapter.callBids(bidRequest, addBidResponse, done, ajax);
-    } else {
-      utils.logError(`Adapter trying to be called which does not exist: ${bidRequest.bidderCode} adaptermanager.callBids`);
-    }
-  });
-}
+  if (bidRequests.length) {
+    let ajax = ajaxBuilder(bidRequests[0].timeout);
+    bidRequests.forEach(bidRequest => {
+      bidRequest.start = new Date().getTime();
+      // TODO : Do we check for bid in pool from here and skip calling adapter again ?
+      const adapter = _bidderRegistry[bidRequest.bidderCode];
+      if (adapter) {
+        utils.logMessage(`CALLING BIDDER ======= ${bidRequest.bidderCode}`);
+        events.emit(CONSTANTS.EVENTS.BID_REQUESTED, bidRequest);
+        bidRequest.doneCbCallCount = 0;
+        let done = doneCb(bidRequest.bidderRequestId);
+        adapter.callBids(bidRequest, addBidResponse, done, ajax);
+      } else {
+        utils.logError(`Adapter trying to be called which does not exist: ${bidRequest.bidderCode} adaptermanager.callBids`);
+      }
+    });
+  } else {
+    utils.logWarn('callBids executed with no bidRequests');
```

Comment:
Maybe add a hint in the log here that it could be because labels didn't match / didn't exist? Or even better if there was an easy code point to detect that condition (I don't see one).

Reply:
Added the hint